### ISSUE TITLE
Question: What is the difference between lora dreambooth vs lora fine-tuning?

### ISSUE THREAD

**Issue by YukiSakuma** [Linaqruf/kohya-trainer#59]
![s](https://user-images.githubusercontent.com/16185735/216850899-39ab1f34-2009-416d-8deb-a553934de829.PNG)
And what is the recommended  method to choose if one trains an artist style/character/concept?

**Comment by satori7272** [comment]
I would like to download the regularization images. They drastically improved my models. How can I put them in when you took it out of the script?

**Comment by Linaqruf** [comment]
Native:
- flexible for larger dataset and longer training if you want to process 1000-100k dataset
- is a traditional fine-tuning and common training method in Machine Learning
- you can train with both captions and tags by defining JSON file 
- not training text encoder by default
- bucketing and cache latents using separate script
- you don't need to set instance prompt, so useful for style training
- you dont need to set <number_of_repeats>_<token> <class> to your foldername like dreambooth

Dreambooth
- for short training
- is a native fine-tuning but for specific use, such as training object or style
- `prior_loss_weight = 1.0`
- training with text encoder by default
- use dreambooth when you want to avoid training with captions (but you can train with captions in kohyas)
- you're training with instance token in dreambooth, eg. `sksbiden` for training Joe Biden dreambooth, and you need to write `sksbiden` as a prompt 
- you can only train with 1 type of caption at a time, e.g. only .txt or .caption
- useful for multi-styled training e.g. Nitrosocke Diffusion, you can train with multiple folder and object in 1 session
- you can use reg images to prevent overfitting 

Use dreambooth when u wanna train on style or object
Use native when u wanna 'resuming' a model training with your own dataset

**Comment by Linaqruf** [comment]
> I would like to download the regularization images. They drastically improved my models. How can I put them in when you took it out of the script?

My notebook still create reg_images folder, the only i took of is a cell for generate and download them directly to reg folder

however you can copy or reg images to reg folder created by this cell
![image](https://user-images.githubusercontent.com/50163983/217428336-462e0e6a-0a26-483d-823d-8f7d37580533.png)

or download them using this cell and set `extract to` to your reg folder path
![image](https://user-images.githubusercontent.com/50163983/217428465-b989b709-5511-43a0-afa3-71b7e5a2784a.png)

**Comment by dimitri-vs** [comment]
@Linaqruf Sorry but its still not clear to me, there is currently four notebooks:

1) Kohya LoRA Dreambooth for LoRA Training (**Dreambooth method**)	
2) Kohya LoRA Fine-Tuning for LoRA Training (**Fine-tune method**)	
3) Kohya Trainer for **Native Training**	
4) Kohya Dreambooth for **Dreambooth Training**

It seems like you compared the 1) Kohya LoRA Dreambooth vs 3) Kohya Trainer for Native Training

But I think the question is if you want to do LoRA training, what is the difference between using 1) Kohya LoRA Dreambooth and 2) Kohya LoRA Fine-Tuning

**Comment by Linaqruf** [comment]
That the same thing, if you want me to specify it, then:
LoRA finetuning:
- Mainly using captions to train
- Using `--in_json` to input captions and tags
- You can use bucketing whle training but in LoRA Finetuning, you can run separate script for bucketing, that is `prepare_buckets_and_latents`, same feature but it will save the latents as `.npz` instead of caching it. Great for resuming training
- Buckets and Latents information saved to where you stored the captions and tags, `meta_lat.json`
- Can declare `--dataset_repeats` without renaming train folder 

LoRA Dreambooth
- Mainly using token to train
- it will ignore token if you put captions, 
- Easy for multi-styled learning, like you can add new token by create new folder and train with it
- `prior_loss_weight = 1.0`
- You can use reg images for data balancing

Basically the same thing.
https://github.com/kohya-ss/sd-scripts/blob/main/train_network.py
![image](https://user-images.githubusercontent.com/50163983/218310601-e1391219-ecc3-4202-aa57-079339f5d8e9.png)

**Comment by YukiSakuma** [comment]
> * it will ignore token if you put captions,

Is token the `concept_name` in the colab notebook? What if you trained with captions but you also added your token like `sksneurosama` in the captions .txt? Will it still be a trigger word?

**Comment by Linaqruf** [comment]
> > * it will ignore token if you put captions,
> 
> Is token the `concept_name` in the colab notebook? What if you trained with captions but you also added your token like `sksneurosama` in the captions .txt? Will it still be a trigger word?

yes, it still be a trigger word, it just not used as token for learning anymore
